hyperspy/_signals/eds.py

```diff
@@ -175,25 +177,61 @@ def sum(self, axis=None, out=None):
             return s
     sum.__doc__ = Signal1D.sum.__doc__
 
-    def rebin(self, new_shape, out=None):
-        new_shape_in_array = []
-        for axis in self.axes_manager._axes:
-            new_shape_in_array.append(
-                new_shape[axis.index_in_axes_manager])
-        factors = (np.array(self.data.shape) /
-                   np.array(new_shape_in_array))
-        s = super().rebin(new_shape, out=out)
-        s = out or s
-        # modify time per spectrum
-        this_md = self.metadata
-        that_md = s.metadata
-        keys = ("Acquisition_instrument.SEM.Detector.EDS.live_time",
-                "Acquisition_instrument.TEM.Detector.EDS.live_time")
-        for key in keys:
-            if key in this_md:
-                that_md.set_item(key, this_md.get_item(key) * np.prod(factors))
-        return s
-    sum.__doc__ = Signal1D.sum.__doc__
+    def rebin(self, new_shape=None, scale=None, crop=True, out=None):
+        """
+        %s
+        """
+
+    #Series of if statements to check that only one out of new_shape or scale
+    #has been given. New_shape is then converted to scale. If both or neither
+    #are given the function raises and error and wont run.
+
+        if new_shape == None and scale == None:
+            raise ValueError("One of new_shape, or scale must be specified")
+        elif new_shape != None and scale != None:
+            raise ValueError("Only one out of new_shape or scale should be specified.\
+                            Not both.")
+        elif new_shape != None:
+            scale = []
+            for i, axis in enumerate(self.data.shape):
```

Comment:
This does not check that `new_shape` length is appropriate. 

I assume we require that full new shape is specified? I.e with signal shape `(5, 3, 2)` and `new_shape=[2, 1]` it fails? 
Alternatively we could assume that the untouched shapes are kept constant, or any surplus shape is ignored.

In the first case (fails) the code should be:
```python
#...
elif new_shape is not None:
    if len(new_shape) != len(self.data.shape):
        raise ValueError("Not enough dimensions in the new shape")
    scale = [datashape/newshape for datashape, newshape in zip(self.data.shape, new_shape)]
```

If we "pad" with ones and ignore the unnecessary, the code should be:
```python
elif new_shape is not None:
    lns = len(new_shape)
    lds = len(self.data.shape)
    if lns < lds:
        new_shape = tuple(new_shape) + (1,)*(lds -lns)
    elif lns > lds:
        new_shape = new_shape[:lds]
    scale = [datashape/newshape for datashape, newshape in zip(self.data.shape, new_shape)]
```